src/async_impl/request.rs

```diff
@@ -136,6 +137,126 @@ impl Request {
         &mut self.version
     }
 
+    /// Get the `gzip` option on this request.
+    #[inline]
+    pub fn gzip(&self) -> Option<&bool> {
```

Comment:
This is my gut feeling: for now, there's less need to provide all these getters on the `Request` type. I would just put the equivalent `gzip(bool)` functions on the `RequestBuilder`.

(Hm, with all these possible options, it occurs to me that a user might just want to say `no_(de)compression()` on either builder, to indicate never use any of the decoders...)

Reply:
> I would just put the equivalent `gzip(bool)` functions on the `RequestBuilder`.

I'm fine with this change. Let me add those functions to `RequestBuilder` first and gather some feedback.